### PR TITLE
Add a GetTextSynchronously method for features that need it.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jasonmalinowski @dotnet/roslyn-ide

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs [hunk 1]
```diff
@@ -107,9 +107,8 @@ protected override void FormatAndApply(Document document, int position, Cancella
         protected override string GetEndingString(Document document, int position, CancellationToken cancellationToken)
         {
             // prepare expansive information from document
```
Comment (Pilchie) [inline src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs]: Might as well fix spelling while you're here.

File: src/Workspaces/Core/Portable/Workspace/Solution/TextDocument.cs [hunk 2]
```diff
@@ -98,6 +98,11 @@ public Task<SourceText> GetTextAsync(CancellationToken cancellationToken = defau
             return this.GetDocumentState().GetTextAsync(cancellationToken);
         }
 
+        internal SourceText GetTextSynchronously(CancellationToken cancellationToken)
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/Workspace/Solution/TextDocument.cs]: Doc comment about when you would use this over `GetTextAsync`.

**Comment by CyrusNajmabadi** [comment]
This needs more work.  Async IO is deeply ingrained in the workspace.

**Pull request closed without merging**